src/main/java/de/tum/in/www1/artemis/domain/lecture/AttachmentUnit.java

```diff
@@ -26,6 +29,10 @@ public class AttachmentUnit extends LectureUnit {
     @Column(name = "description")
     private String description;
 
+    @Enumerated(EnumType.STRING)
+    @Column(name = "pyris_ingestion_state")
```

Comment:
This change couples the IRIS and the lecture modules quite a bit on the database level. I was wondering if persisting the state is really necessary.

It does not really hurt to store it in the database, but as we want to decouple the modules more strictly in the future, I am wondering if this change is a good idea 🤔

@bassner @MaximilianAnzinger what do you think?

Reply:
My view:

We could use an HTTP query to check Pyris for the ingestion status instead of storing it in Artemis when it's unknown. Since Pyris is responsible for ingestion, it should report the status. This would prevent displaying incorrect states, such as if the data in Weaviate has been changed, deleted, or rolled back, or if Pyris has been restarted during ingestion. In such cases, Artemis might continuously show "In Progress" even though nothing is actually happening, or "Done" even though the data is not really there.

Pyris can still proactively report updates during ingestion, and we could distribute them to the frontend clientsvia websocket. If necessary, we can further optimize this by caching the ingestion states in Hazelcast (or Redis in the future), although that might not be needed as ingestion is not a frequent use-case (rather one time per lecture or something).

Reply:
TLDR: I agree with Patrick here.

One additional HTTP query should not be too expensive and eliminates invalid states (as stated above).
Also consider the coupling this introduces for your upcoming modularization. Pyris for sure should act on the Lecture-Module, however, it shouldn't store anything within the Lecture-Module itself.
If you opt for the solution to persist this value, IMO it should be done in a separate table.

Reply:
Okay then I will go for making everything through Api calls thank you for the feedback.